OpenRA.Mods.Common/Traits/Buildings/Dock.cs

```diff
@@ -0,0 +1,131 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits.Render;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public abstract class DockInfo : TraitInfo, Requires<WithSpriteBodyInfo>, IAcceptResourcesInfo
+	{
+		[Desc("Actual harvester facing when docking.")]
+		public readonly WAngle DockAngle = WAngle.Zero;
+
+		[Desc("Docking cell relative to top-left cell.")]
+		public readonly CVec DockOffset = CVec.Zero;
+
+		[Desc("Does the refinery require the harvester to be dragged in?")]
+		public readonly bool IsDragRequired = false;
+
+		[Desc("Vector by which the harvester will be dragged when docking.")]
+		public readonly WVec DragOffset = WVec.Zero;
+
+		[Desc("In how many steps to perform the dragging?")]
+		public readonly int DragLength = 0;
+	}
+
+	// TODO: turn this into a trait
+	public abstract class Dock : INotifyCreated, ITick, INotifySold, INotifyCapture,
```

Comment:
Fwiw the naming convention for `abstract` classes is including `Base` (like `CellLayerBase`). How fast will this class be turned into a non-abstract one? Can we remove the TODO then (to reduce noise)?

Reply:
It's going to take quite a lot of PR's, and by our current review rate it I doubt that it's gonna even happen this year

Reply:
As it's a temporarily abstract class should we really give it `base`? This convention also seems to be quite poorly practiced, most abstract classes I've interacted with don't have it

Reply:
Ok, then leave it as is for now.